source/grid/grid_generator.cc

```diff
@@ -28,11 +29,1219 @@
+                const double m = 1.0 * digit_0 / 100; // max. chamber
+                const double p = 1.0 * digit_1 / 10; // location of max. chamber
+                const double x = i * 1 / (1.0 * number_points - 1);
+
+                const double y_c =
+                  (x <= p) ? m / std::pow(p, 2) * (2 * p * x - std::pow(x, 2)) :
+                             m / std::pow(1 - p, 2) *
+                               ((1 - 2 * p) + 2 * p * x - std::pow(x, 2));
+
+                const double dy_c = (x <= p) ?
+                                      2 * m / std::pow(p, 2) * (p - x) :
+                                      2 * m / std::pow(1 - p, 2) * (p - x);
+
+                const double y_t =
+                  5 * t *
+                  (0.2969 * std::pow(x, 0.5) - 0.126 * x -
+                   0.3516 * std::pow(x, 2) + 0.2843 * std::pow(x, 3) -
+                   0.1036 * std::pow(x, 4)); // half thicknes at a position x
+
+                const double theta = std::atan(dy_c);
+
+                if (is_upper)
+                  naca_points.emplace_back(x - y_t * std::sin(theta),
+                                           y_c + y_t * std::cos(theta));
+                else
+                  naca_points.emplace_back(x + y_t * std::sin(theta),
+                                           y_c - y_t * std::cos(theta));
+              }
+
+          return naca_points;
+        }
+
+
+
+        /**
+         * Set airfoil length (i.e. chord length) to a desired length.
+         * calls function set_airfoil_length() for each vector of the array
+         *
+         * @param[in] input Array containing upper and lower vector.
+         * @param[in] desired_len Indicates desired length of input vector.
+         * @return output Array of two vectors with desired length.
+         */
+        static std::array<std::vector<Point<2>>, 2>
+        set_airfoil_length(const std::array<std::vector<Point<2>>, 2> &input,
+                           const double desired_len)
+        {
+          std::array<std::vector<Point<2>>, 2> output;
+          output[0] = set_airfoil_length(input[0], desired_len);
+          output[1] = set_airfoil_length(input[1], desired_len);
+
+          return output;
+        }
+
+        /**
+         * Set airfoil length (i.e. chord length) to a desired length.
+         *
+         * @param[in] input Vector containing upper or lower points.
+         * @param[in] desired_len Indicates desired length of input vector.
+         * @return output Scaled vector with desired length.
+         * */
+        static std::vector<Point<2>>
+        set_airfoil_length(const std::vector<Point<2>> &input,
+                           const double                 desired_len)
+        {
+          std::vector<Point<2>> output(input.size());
+
+          const auto scale = desired_len / input.front().distance(input.back());
+
+          for (unsigned i = 0; i < input.size(); i++)
+            output[i] = input[i] * scale;
```

Comment:
I think you can write this whole function simpler by saying
```
  std::vector<Point<2>> output = input;
  const double scale = ...;
  for (auto &x : output)
    x *= scale;
  return output;
```
It's the same length in # of lines, but it's a bit easier to read what is happening.